aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -453,13 +454,39 @@ bool check_runtime_disabled_mem_efficient(sdp_params params, bool debug) {
   return true;
 }
 
+template <int Major, int Minor>
+struct SMVersion {
+  static constexpr int major = Major;
+  static constexpr int minor = Minor;
+  constexpr SMVersion() = default;
+};
+
+/**
+ * Checks if the current CUDA device architecture is inclusively within the specified range.
+ *
+ * @param lower_bound The lower bound of the CUDA device architecture range.
+ * @param upper_bound The upper bound of the CUDA device architecture range.
+ * @param params The parameters for the current operation.
+ * @return True if the current CUDA device architecture is within the specified range, false otherwise.
+ */
+template <typename lower_bound, typename upper_bound>
+bool check_sm_version(sdp_params params) {
+  auto dprops = at::cuda::getCurrentDeviceProperties();
+  bool is_gte_lower_bound = dprops->major > lower_bound::major ||
+      (dprops->major == lower_bound::major &&
+       dprops->minor >= lower_bound::minor);
+  bool is_lte_upper_bound = dprops->major < upper_bound::major ||
+      (dprops->major == upper_bound::major &&
+       dprops->minor <= upper_bound::minor);
+  return is_gte_lower_bound && is_lte_upper_bound;
+}
+
 bool check_gpu_sm75_or_greater(sdp_params params, bool debug) {
   // Check that the gpu is capable of running flash attention
-  auto dprops = at::cuda::getCurrentDeviceProperties();
-  bool is_sm75 = dprops->major == 7 && dprops->minor == 5;
-  bool is_sm8x = dprops->major == 8 && dprops->minor >= 0;
-  bool is_sm90 = dprops->major == 9 && dprops->minor == 0;
-  if (!(is_sm90 || is_sm8x || is_sm75)) {
+  using sm75 = SMVersion<7, 5>;
+  using sm90 = SMVersion<9, 0>;
+  if (!check_sm_version<sm75, sm90>(params)) {
+    auto dprops = at::cuda::getCurrentDeviceProperties();
     if (debug) {
       TORCH_WARN(
           "Flash attention only supports {sm75, sm8x, sm90} gpu architectures. Attempting to run on a sm ",
```

Comment:
I think you'll also need to update this warning. Are we sure it works on the full range of [sm75, sm90]?

Reply:
yeah I didn't actualy change any of this logic here, this is just updating to use the new helper

Reply:
Hm, yeah I guess there aren't really any SM 7.7 or so GPUs. There is an SM 7.2. At least according to https://developer.nvidia.com/cuda-gpus . The previous check matched on SM7.5 exactly and then [SM8.0, SM9.0] but now it's [SM7.5, SM9.0].

Reply:
@drisspg - can you update the warning message to show that we're checking this range (just like the other warnings)?